frontend/javascripts/dashboard/dataset/sample_datasets_modal.js

```diff
@@ -0,0 +1,146 @@
+// @flow
+import { Spin, Button, Modal, List } from "antd";
+import React, { useState, useEffect } from "react";
+import _ from "lodash";
+
+import {
+  getSampleDatasets,
+  triggerSampleDatasetDownload,
+  getDatastores,
+} from "admin/admin_rest_api";
+import { useInterval } from "libs/react_helpers";
+import { handleGenericError } from "libs/error_handling";
+import type { APIDataStore, APISampleDataset } from "admin/api_flow_types";
+
+type Props = {
+  destroy: () => void,
+  onOk?: () => any,
+  organizationName: string,
+};
+
+function useDatastores(): [Array<APIDataStore>] {
+  const [datastores, setDatastores] = useState([]);
+
+  const fetchDatastores = async () => {
+    const fetchedDatastores = await getDatastores();
+    setDatastores(fetchedDatastores);
+  };
+
+  useEffect(() => {
+    fetchDatastores();
+  }, []);
+  return [datastores];
+}
+
+function useSampleDatasets(
+  datastore,
+  organizationName,
+  pendingDatasets,
+  setPendingDatasets,
+): [Array<APISampleDataset>, Array<string>, () => Promise<void>] {
+  const [datasets, setDatasets] = useState([]);
+  const [failedDatasets, setFailedDatasets] = useState([]);
+
+  const fetchSampleDatasets = async () => {
+    if (datastore == null) return;
+    const sampleDatasets = await getSampleDatasets(datastore.url, organizationName);
+    setDatasets(sampleDatasets);
+
+    // Datasets that were pending, but are now available again, failed to download
```

Comment:
Yes, since the download is running in the backend, it is not easy to communicate a “failed” state (it would have to expire). Hence it simply turns back to available. And the front-end treats that as failed for this session.